src/GraphQL.Tests/Utilities/FederatedSchemaBuilderTests.cs

```diff
@@ -120,5 +121,34 @@ public void entity_query()
                 _.ExpectedResult = expected;
             });
         }
+
+        [Fact]
+        public void input_types_and_types_without_key_directive_are_not_added_to_entities_union()
```

Comment:
@joemcbride I haven’t had time yet to sort out federated schemas. Does this test seem right to you?